### PR TITLE
API Assertion error errata

### PR DESCRIPTION
**Pull request by akhil-jha**
1. Adding two method in robottelo/hosts.py for getting and attaching pool_id.
2. Adding `wait_for_tasks` for `Actions::Katello::Host::UploadProfiles`
3. Moving from module_org to function where there is no need of RH repo.
4. Merging two tests `test_positive_filter_by_cve` and `test_positive_sort_by_issued_date`
5. A module fixture to setup content for rhel6 and removing the corresponding extra code.
6. Reassessing case importance

### CODE REVIEW

**Comment by akhil-jha** [comment]
trigger: test-robottelo
pytest: --importance High tests/foreman/api/test_errata.py

**Comment by akhil-jha** [comment]
trigger: test-robottelo
pytest: --importance High tests/foreman/api/test_errata.py

File: robottelo/hosts.py [hunk 1]
```diff
@@ -257,6 +257,24 @@ def subscription_manager_status(self):
     def subscription_manager_list(self):
         return self.execute('subscription-manager list')
 
+    def subscription_manager_get_pool(self, sub_list=[]):
+        pool_id = []
```
Comment (omkarkhatavkar) [inline robottelo/hosts.py:261]: it should not be singular, could be `pool_id_list` or pool_ids

File: robottelo/hosts.py [hunk 2]
```diff
@@ -257,6 +257,24 @@ def subscription_manager_status(self):
...
+        for sub in sub_list:
+            result = self.execute(
+                f'subscription-manager list --available --pool-only --matches="{sub}"'
+            )
+            result = result.stdout
+            result = result.split('\n')
+            result = ' '.join(result).split()
+            pool_id.append(result)
```
Comment (omkarkhatavkar) [inline robottelo/hosts.py:269]: `list.append(result.strip().split('\n'))` is this in one line works for you ? did you faced issue extra empty strings ?

File: tests/foreman/api/test_errata.py [hunk 3]
```diff
@@ -721,13 +651,16 @@ def test_positive_get_diff_for_cv_envs():
...
     both_cvvs_errata = next(
-        errata for errata in result['results'] if errata['errata_id'] == constants.REAL_0_ERRATA_ID
+        errata
+        for errata in result['results']
+        if errata['errata_id'] in constants.FAKE_9_YUM_SECURITY_ERRATUM
     )
     assert {cvv.id for cvv in cvvs} == set(both_cvvs_errata['comparison'])
 
 
+@pytest.mark.skip_if_open("BZ:2013093")
```
Comment (omkarkhatavkar) [inline tests/foreman/api/test_errata.py:663]: :bz: tag can be added in the test metadata as well

File: robottelo/hosts.py (hunk 2 above)
Comment (akhil-jha) [reply robottelo/hosts.py]: I initially tried that, but I was getting TypeError, I had to resort to the long format.

**Comment by akhil-jha** [comment]
trigger: test-robottelo
pytest: --importance High tests/foreman/api/test_errata.py

**Comment by akhil-jha** [comment]
trigger: test-robottelo
pytest: --importance High tests/foreman/api/test_errata.py

**Comment by akhil-jha** [comment]
trigger: test-robottelo
pytest: --importance High tests/foreman/api/test_errata.py

**Comment by akhil-jha** [comment]
trigger: test-robottelo
pytest: --importance High tests/foreman/api/test_errata.py

**Comment by swadeley** [comment]
trigger: test-robottelo
pytest: --importance High tests/foreman/api/test_errata.py

**Comment by akhil-jha** [comment]
trigger: test-robottelo
pytest: --importance High tests/foreman/api/test_errata.py

**Comment by akhil-jha** [comment]
trigger: test-robottelo
pytest: --importance High tests/foreman/api/test_errata.py

**Comment by akhil-jha** [comment]
trigger: test-robottelo
pytest: --importance High tests/foreman/api/test_errata.py

**Comment by akhil-jha** [comment]
trigger: test-robottelo
pytest: --importance High tests/foreman/api/test_errata.py

**Comment by akhil-jha** [comment]
trigger: test-robottelo
pytest: --importance High tests/foreman/api/test_errata.py

**Comment by swadeley** [comment]
> I wonder will the way JobInvocation is called be more consistent or less consistent after this prosed change?

Hi, you still have two methods of Job Invocation:
I see you replaced two `default_sat.api.JobInvocation().run` with `JobInvocation.create`
but one `default_sat.JobInvocation().run` was  changed to `default_sat.api.JobInvocation().run`

Not a blocker.

**Review by swadeley — APPROVED** [review]
ACK

**Comment by akhil-jha** [comment]
trigger: test-robottelo
pytest: --importance High tests/foreman/api/test_errata.py

**Comment by akhil-jha** [comment]
trigger: test-robottelo
pytest: --importance High tests/foreman/api/test_errata.py

File: robottelo/hosts.py [hunk 4]
```diff
@@ -257,6 +257,24 @@ def subscription_manager_status(self):
...
 
+    def subscription_manager_get_pool(self, sub_list=[]):
+        pool_ids = []
+        for sub in sub_list:
+            result = self.execute(
+                f'subscription-manager list --available --pool-only --matches="{sub}"'
+            )
+            result = result.stdout
+            result = result.split('\n')
+            result = ' '.join(result).split()
```
Comment (jyejare) [inline robottelo/hosts.py:268]: Can be combined into one:
```suggestion
            result = result.stdout
            result = result.split('\n')
            result = ' '.join(result.stdout.split('\n')).split()
```
Non blocking Comment!

File: robottelo/hosts.py [hunk 5]
```diff
@@ -257,6 +257,24 @@ def subscription_manager_status(self):
     def subscription_manager_list(self):
         return self.execute('subscription-manager list')
 
+    def subscription_manager_get_pool(self, sub_list=[]):
```
Comment (jyejare) [inline robottelo/hosts.py:260]: Add a one docstring, it could be small as well.

File: robottelo/hosts.py [hunk 6]
```diff
@@ -257,6 +257,24 @@ def subscription_manager_status(self):
...
+            result = self.execute(
+                f'subscription-manager list --available --pool-only --matches="{sub}"'
+            )
+            result = result.stdout
+            result = result.split('\n')
+            result = ' '.join(result).split()
+            pool_ids.append(result)
+        return pool_ids
+
+    def subscription_manager_attach_pool(self, pool_list=[]):
```
Comment (jyejare) [inline robottelo/hosts.py:272]: Add a one docstring, it could be small as well.

File: tests/foreman/api/test_errata.py [hunk 7]
```diff
@@ -17,23 +17,28 @@
...
 
 from robottelo import constants
+from robottelo import manifests
 from robottelo.api.utils import enable_rhrepo_and_fetchid
 from robottelo.api.utils import promote
+from robottelo.api.utils import upload_manifest
+from robottelo.api.utils import wait_for_tasks
 from robottelo.cli.factory import setup_org_for_a_custom_repo
 from robottelo.cli.factory import setup_org_for_a_rh_repo
+from robottelo.cli.job_invocation import JobInvocation
```
Comment (jyejare) [inline tests/foreman/api/test_errata.py:35]: Dont we have API entity in nailgun for this ?

File: tests/foreman/api/test_errata.py [hunk 8]
```diff
@@ -200,21 +204,31 @@ def test_positive_install_in_hc(module_org, activation_key, custom_repo, rh_repo
...
+        JobInvocation.create(
+            {
                 'feature': 'katello_errata_install',
-                'inputs': {'errata': f'{CUSTOM_REPO_ERRATA_ID}'},
-                'targeting_type': 'static_query',
-                'search_query': f'host_collection_id = {host_collection.id}',
-                'organization_id': module_org.id,
+                'inputs': f'errata=\'{CUSTOM_REPO_ERRATA_ID}\'',
+                'search-query': host_collection_query,
+                'organization-id': f'{module_org.id}',
```
Comment (jyejare) [inline tests/foreman/api/test_errata.py:213]: Any reason you switched to CLI here ?

File: tests/foreman/api/test_errata.py [hunk 9]
```diff
@@ -286,15 +308,24 @@ def test_positive_install_multiple_in_host(
     applicable_errata_count = host.content_facet_attributes['errata_counts']['total']
     assert applicable_errata_count > 1
     rhel7_contenthost.add_rex_key(satellite=default_sat)
-    for errata in settings.repos.yum_9.errata[:2]:
-        default_sat.JobInvocation().run(
-            data={
+    for errata in settings.repos.yum_9.errata[1:4]:
+        timestamp = (datetime.utcnow()).strftime('%Y-%m-%d %H:%M')
+        JobInvocation.create(
```
Comment (jyejare) [inline tests/foreman/api/test_errata.py:313]: Same here, via CLI if we have faster API.

File: tests/foreman/api/test_errata.py [hunk 10]
```diff
@@ -513,8 +476,62 @@ def test_positive_filter_by_envs(module_org):
     assert len(errata_library) > len(errata_env)
 
 
+@pytest.fixture(scope='module')
+def setup_content_rhel6():
```
Comment (jyejare) [inline tests/foreman/api/test_errata.py:480]: A small docstring should help !

Also, you can move this to global fixtures !

File: tests/foreman/api/test_errata.py [hunk 11]
```diff
@@ -513,8 +476,62 @@ def test_positive_filter_by_envs(module_org):
...
+    org = entities.Organization().create()
```
Comment (jyejare) [inline tests/foreman/api/test_errata.py:481]: You should be able to use many of the global fixtures here (like here module_org | func_org), you can make a plan to move this to global fixtures / use global fixtures. 

You can raise a followup PR for improvements !

File: tests/foreman/api/test_errata.py [hunk 12]
```diff
@@ -513,8 +476,62 @@ def test_positive_filter_by_envs(module_org):
...
+
+    ak = entities.ActivationKey(content_view=cv, organization=org, environment=lce).create()
+
+    sub_list = [DEFAULT_SUBSCRIPTION_NAME, host_tools_product.name, custom_product.name]
+    for sub_name in sub_list:
+        subscription = entities.Subscription(organization=org).search(
+            query={'search': f'name="{sub_name}"'}
+        )[0]
+        ak.add_subscriptions(data={'subscription_id': subscription.id})
+    return ak, org, sub_list
```
Comment (jyejare) [inline tests/foreman/api/test_errata.py:530]: Returning dictionary instead will look sane ! 

But its not a blocking comment !

**Review by jyejare — APPROVED** [review]
ACK with mostly Non Blocking comments !

@akhil-jha Feel free to merge yourself since now u ll have 2 ACKs without my non-blocking comments.

File: robottelo/hosts.py (hunk 4 above)
Comment (akhil-jha) [reply robottelo/hosts.py:271]: I tried to be a little explicit for better readability. 
I will have a followup PR for all the small improvements.

File: robottelo/hosts.py (hunk 6 above)
Comment (akhil-jha) [reply robottelo/hosts.py:275]: Done

File: tests/foreman/api/test_errata.py (hunk 11 above)
Comment (akhil-jha) [reply tests/foreman/api/test_errata.py:487]: Will do in the followup PR

File: tests/foreman/api/test_errata.py (hunk 12 above)
Comment (akhil-jha) [reply tests/foreman/api/test_errata.py:536]: Followup PR

File: tests/foreman/api/test_errata.py (hunk 10 above)
Comment (akhil-jha) [reply tests/foreman/api/test_errata.py:480]: Doc string added.
Followup PR for making it global

File: tests/foreman/api/test_errata.py (hunk 7 above)
Comment (akhil-jha) [reply tests/foreman/api/test_errata.py:35]: We do. I was getting some flakiness time out in the polling after the Job from API. As the test is focused upon getting the errata installed irrespective of how job is invoked, I choose to use the CLI instead.

File: tests/foreman/api/test_errata.py (hunk 8 above)
Comment (akhil-jha) [reply tests/foreman/api/test_errata.py:213]: Stated above.

File: tests/foreman/api/test_errata.py (hunk 9 above)
Comment (akhil-jha) [reply tests/foreman/api/test_errata.py:313]: Stated above.

**Comment by akhil-jha** [comment]
@jyejare , I will have a PR up for the improvements you've mentioned.

**Comment by akhil-jha** [comment]
Merging.

**Pull request merged**